src/main.rs

```diff
@@ -1734,8 +1799,9 @@ async fn main() {
                 let format = setup.format;
                 let backend = setup.backend;
                 let device = setup.device.clone();
+                let sample_rate = player_config.sample_rate.as_u32();
                 let player = Player::new(player_config, session.clone(), soft_volume, move || {
-                    (backend)(device, format)
+                    (backend)(device, format, sample_rate)
```

Comment:
Could we avoid having to duplicate the sample rate thing here? (as in: The sample rate is part of `player_config` and still has to be extracted and passed into the backend builder separately.) We could e.g. pass the `sample_rate` via a closure parameter to the builder, which could then pass it along? But that doesn't sound so good either.

Reply:
That's the sample rate as a u32 not the sample_rate enum.

Reply:
Previously the backends used the SAMPLE_RATE constant which was a u32. This just passes in the sample rate as a u32 since it's no longer constant.